packages/database/src/migrations/20220630013151-LinkSyncServiceToSurvey-modifies-schema.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.runSql(
+    `DROP TRIGGER IF EXISTS data_service_sync_group_trigger ON data_service_sync_group`,
+  );
+
+  await db.runSql(`ALTER TABLE data_service_sync_group RENAME COLUMN code TO data_group_code`);
```

Comment:
Actually I don't understand how this works. We can't change code here without updating DataBroker.

https://github.com/beyondessential/tupaia/blob/dev/packages/data-broker/src/DataBroker.js#L82

Reply:
Yeah honestly I found this rename a bit confusing as well.
It seems like a dssg has a `data_group_code` which corresponds to a survey code in Tupaia. But should the `syncGroupCode` in the `config` actually just be the `code` of this dssg? I made that mistake a couple times when writing logic around this, as it seems really weird to me that dssg's don't have a code of their own atm...

I'd suggest changing so that:
```
current:
id  |  data_group_code | service_type |         config             | sync_cursor
-----------------------------------------------------------------------------------
123 |    LEMIS_FQS1    |      kobo    | { syncGroupCode: FQS1, ...} | 10-10-2020

becomes:
id  | code |  data_group_code | service_type | config | sync_cursor
------------------------------------------------------------------------
123 | FQS1 |     LEMIS_FQS1   |      kobo    | { ...} | 10-10-2020

```
Then when pulling from data-broker, you do:
```
const koboData = await dataBroker.pull(
  {
    code: 'FQS1',
    type: 'SYNC_GROUP',
  }
);
```
Thoughts?

You could possibly argue that the `koboSurveyCode` (which exists in the `config`) should also have its own column, but I think given that this table is meant to be agnostic in regards to the `service_type` its appropriate for it to be in the config.

Reply:
Also, just wanted to flag a slight concern I have around how often we rely on pseudo foriegn key relationships. Here the `data_group_code` _**must**_ correspond to a survey code. But we don't enforce that in any way via foreign key constraints. It's quiet common for us to do this in Tupaia, but it always feels a bit risky to me... Do you think I ought to add a constraint here?

Reply:
Discussed, like the code column, FK not necessary but worth a chat in the future.